packages/devtools_app_shared/lib/src/service/isolate_manager.dart

```diff
@@ -104,7 +104,16 @@ final class IsolateManager with DisposerMixin {
     // current isolate, in which case the breakpoint manager will resume it
     // after setting breakpoints):
     if (selectedIsolate.value?.id != isolateRef.id) {
-      await resumeIsolateIfPaused(isolateRef);
+      final isWebApp =
+          serviceConnection.serviceManager.connectedApp?.isDartWebAppNow ??
+              false;
+      // TODO(bug link) Once we can reliably determine whether a web app is
+      // paused, then switch to resumeIsolateIfPaused.
+      if (isWebApp) {
+        await resumeIsolate(isolateRef);
+      } else {
+        await resumeIsolateIfPaused(isolateRef);
+      }
```

Comment:
In DAP we tend to just always resume and discard the error if it was not paused. Checking like this can have race conditions if an isolate is unpaused between you checking and the resume request being processed (for example because the user hit resume, or the isolate was only paused temporarily while breakpoints were being set or something).